cumulus/pallets/xcmp-queue/src/lib.rs

```diff
@@ -952,7 +952,7 @@ impl<T: Config> SendXcm for Pallet<T> {
 
 /// Checks that the XCM is decodable with `MAX_XCM_DECODE_DEPTH`.
 ///
-/// Note that this uses the limit of the sender - not the receiver. It it best effort.
+/// Note that this uses the limit of the sender - not the receiver. It is a best effort.
 pub(crate) fn validate_xcm_nesting(xcm: &VersionedXcm<()>) -> Result<(), ()> {
```

Comment:
@ggwpez I see that you added `validate_xcm_nesting` [here](https://github.com/paritytech/polkadot-sdk/pull/1246/files#diff-31d504e3776a85fc47e3c11e673819cc87bdaf712bb613ea366360fe8b8a1786R929) but only for `XcmpQueue`. Don't you remember if there was any reason not to add it also to the [`ParentAsUmp`](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/primitives/utility/src/lib.rs#L70-L72) or [`ChildParachainRouter`](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/runtime/common/src/xcm_sender.rs#L122)?

And another question, is that [`defensive_assert`](https://github.com/paritytech/polkadot-sdk/pull/1246/files#diff-31d504e3776a85fc47e3c11e673819cc87bdaf712bb613ea366360fe8b8a1786R945) still needed? I think, that check in `fn validate` is enough, could we remove it?

Reply:
> I did not think about the fact that other delivery protocols could encounter the same issue, so i only added it for HRMP outbound messages.

@ggwpez I created a separate [PR](https://github.com/paritytech/polkadot-sdk/pull/4236) for adding `validate_xcm_nesting` elsewhere, but for completeness, what kind of issue was that? I mean, why does it only affect HRMP and not DMP or UMP? Could you please provide more info to that PR (if possible)?